lib/iris/fileformats/pp.py

```diff
@@ -1644,10 +1644,6 @@ def __eq__(self, other):
                 if all(attrs):
                     self_attr = getattr(self, attr)
                     other_attr = getattr(other, attr)
-                    if isinstance(self_attr, biggus.NumpyArrayAdapter):
-                        self_attr = self_attr.concrete
-                    if isinstance(other_attr, biggus.NumpyArrayAdapter):
-                        other_attr = other_attr.concrete
```

Comment:
I've removed the checks for lazy arrays, so the data will be loaded during the comparison, which may not be desirable. I don't think the underlying array of a dask array is easily accessible, like with `biggus.NumpyArrayAdapter.concrete`.

It turns out that all the attributes of the proxy object (which is `.concrete` here) are derived from other attributes on the pp field, with the exception of the filename. So if the filename were recorded as an attribute of the PPField the same behaviour as before could be preserved.